File: TechNode.Core/Entities/Product.cs
namespace TechNode.Core.Entities;

public class Product : BaseEntity
{
    public required string Name { get; set; }

    public required string Description { get; set; }

    public decimal Price { get; set; }

    public required string PictureUrl { get; set; }

    public required string Type { get; set; }

    public required string Brand { get; set; }

    public int StockQuantity { get; set; }
    
    public int CategoryId { get; set; }

    public Category Category { get; set; } = null!;

    public ICollection<ProductSpecification> ProductSpecifications { get; set; } = [];
    
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: client/src/app/shared/models/product.ts
export interface Product {
  id: number;
  name: string;
  description: string;
  price: number;
  pictureUrl: string;
  brand: string;
  stockQuantity: number;
  categoryName: string;
  specifications: {
    [key: string]: string;
  };
}

export interface ProductAddRequest {
  name: string;
  description: string;
  price: number;
  pictureUrl: string;
  brand: string;
  stockQuantity: number;
  categoryId: number;
  specifications: {
    [key: string]: string;
  };
}
